std/algorithm/searching.d

```diff
@@ -2960,6 +3035,111 @@ unittest
     }
 }
 
+/**
+Iterates the passed range and returns the maximal element.
+A custom mapping for comparison key can be used with $(D map).
+
+Params:
+    map = custom mapping for the comparison key
+    r = Range from which the maximum will be selected
+
+Returns: The maximal element of the passed-in range.
+
+See_Also:
+    $(XREF max)
+*/
+auto maxElement(alias map = "a", Range)(Range r)
+    if (isInputRange!Range && !isInfinite!Range)
+{
+    return extremum!(map, "a > b")(r);
+}
+
+///
+unittest
+{
+    import std.range: enumerate, iota;
+    assert([2, 1, 4, 3].maxElement == 4);
+    // allows to get the index too
+    assert([2, 1, 4, 3].enumerate.maxElement!"a.value" == tuple(2, 4));
```

Comment:
Needs another example of a custom accessor, something like the assert on line 1279